File: DerivativeSigmoid.py
'''
Function derivative of the sigmoid (logistic function)
Функция производной сигмоиды (логистической функции) на Python может быть реализована следующим образом:
'''
python
import numpy as

def sigmoid(x):
    return 1 / (1 + np.exp(-x))

def sigmoid_derivative(x):
    sig = sigmoid(x)
    return sig * (1 - sig)
'''
В этом примере мы используем библиотеку NumPy для вычисления экспоненты и элемент-по-элементного умножения. Функция sigmoid вычисляет значение сигмоиды для данного входного значения x, а функция sigmoid_derivative вычисляет производную сигмоиды по формуле sig * (1 - sig), где sig - значение сигмоиды.

Примерания:

python
x = 2
print(sigmoid_derivative(x))  # Вывод: 0.10458540350662
Этот код вычисляет производную сигмоиды для x = 2 и выводит.

'''

File: DerivativeSigmoid.js
//реализация функции производной сигмоиды на JavaScript

/*
В этом примере мы используем Math.exp() для вычисления экспоненты и Math.multiply() для элемент-по-элементного умножения.
Функция sigmoid вычисляет значение сигмоиды для данного входного значения x,
а функция sigmoid_derivative вычисляет производную сигмоиды по формуле sig * (1 - sig),
где sig - значение сигмоиды.
*/

function sigmoid(x) {
  return 1 / (1 + Math.exp(-x));
}

function sigmoid_derivative(x) {
  var sig = sigmoid(x);
  return sig * (1 - sig);
}

//example
var x = 2;
console.log(sigmoid_derivative(x)); // Вывод: 0.10458540350662
